src/main/asciidoc/reference/neo4j-repositories.adoc

```diff
@@ -663,5 +663,41 @@ Delete graph entities::
 [NOTE]
 For users coming from versions before `4.2.x`, `Neo4jRepository` has replaced `GraphRepository` but essentially has the same features.
 
+[[reference_programming-model_constructors]]
+=== Persistence constructors
+
+From SDN 5.1 (Lovelace release train), persistence constructors are supported.
+
+It means that SDN will try to find the best constructor to instantiate entities.
+Constructor discovery works as follow :
+
+- look for explicit @PersistenceConstructor annotation on constructor
+- if not found, look for the default constructor
+- if not found, look at any specific constructor defined by the entity.
+
+Values for constructor parameters are determined by looking at parameters names (not parameter types).
+For values to be filled in correctly, the parameter names have to match the entity attributes names.
```

Comment:
1. Doesn't this mean domain classes need to be compiled with `-parameters` flag? Should be mentioned here.
1. How does the test code end up with it? I don't see it anywhere..

Reply:
1. No. Actually when the code in not compiled with -parameters flag, Spring Data falls back to byte code parsing to get argument names.